# Importer les bibliothèque

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Importer les données

In [5]:
#Importer jeu d'entrainement
data_train = pd.read_csv('donnees/train.tsv',sep='\t')
data_train.info()
train = data_train[['headline','text','category']]

#Importer jeu de test
data_test= pd.read_csv('donnees/test.tsv',sep='\t')
test = data_test[['headline','text','category']]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1476 entries, 0 to 1475
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   category  1476 non-null   object
 1   headline  1476 non-null   object
 2   text      1476 non-null   object
 3   url       1476 non-null   object
dtypes: object(4)
memory usage: 46.2+ KB


On garde seulement headlin, text, category parce qu'on doit se baser sur ça pour la classification

# Analyse des données

## Analyse des données générale

In [11]:
data_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1476 entries, 0 to 1475
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   category  1476 non-null   object
 1   headline  1476 non-null   object
 2   text      1476 non-null   object
 3   url       1476 non-null   object
dtypes: object(4)
memory usage: 46.2+ KB


In [7]:

data_train.head()

category                                           headline  \
0    sports         L'Ouganda à l'assaut des "fimbu" de la RDC   
1  business  Stopper la détérioration de l’environnement po...   
2    sports  Coupe d'Afrique des nations 2022 : le sélectio...   
3  business  Tissus africains : pourquoi les teintureries h...   
4  business  Les revenus pendant la pandémie des dix hommes...   

                                                text  \
0  L'Ouganda, placé 79e au classement FIFA le 4 a...   
1  La responsable de la biodiversité des Nations ...   
2  Le sélectionneur de la Sierra Leone, John Keis...   
3  Depuis plus de six siècles, une vaste zone sit...   
4  Pendant la pandémie de coronavirus, la richess...   

                                                 url  
0        https://www.bbc.com/afrique/sports-48730747  
1  https://www.bbc.com/afrique/articles/cevdz4xrp42o  
2        https://www.bbc.com/afrique/region-59881445  
3        https://www.bbc.com/afrique/region-61202726  
4         https://www.bbc.com/afrique/monde-55812018

In [8]:
data_train.describe()

category                                           headline  \
count      1476                                               1476   
unique        5                                               1474   
top      sports  Le souchet, tubercule sauvage devenu "super al...   
freq        350                                                  2   

                                                     text  \
count                                                1476   
unique                                               1474   
top     Le souchet est un amuse-bouche qu’on peut trou...   
freq                                                    2   

                                                      url  
count                                                1476  
unique                                               1474  
top     https://www.bbc.com/afrique/articles/cv2rg2xe4qyo  
freq                                                    2

Le jeu d'entrainement possède 1476 entrée de données.


## Analyse des données utiles pour notre cas

In [12]:
train.info()
#train.describe()
train['category'].value_counts()
#train['category'].value_counts().plot(kind='bar')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1476 entries, 0 to 1475
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   headline  1476 non-null   object
 1   text      1476 non-null   object
 2   category  1476 non-null   object
dtypes: object(3)
memory usage: 34.7+ KB


category
sports        350
business      350
politics      350
health        350
technology     76
Name: count, dtype: int64

Toutes les catégories n'ont pas le meme nombre d'entrée de données.Il sera donc nécessaire de faire un traitement pour équilibrer les données.

Il est important de vérifier qu'il n'y a pas de données manquante


In [13]:
train.isna().sum()

## Pas de valeur manquante

headline    0
text        0
category    0
dtype: int64